feat: Add getting started with AKS guide (#624)
stackabletech/documentation: modules/ROOT/pages/kubernetes/aks.adoc

> = Azure Kubernetes services (AKS)

Author: I got it from here, but let's stick to your suggestion
![image](https://github.com/stackabletech/documentation/assets/29303194/4312fda0-f552-40a8-9a07-120f349e5a3f)

> = Azure Kubernetes services (AKS)

Reviewer: I got it from their docs: https://azure.microsoft.com/en-us/products/kubernetes-service (just for reference)

Automatic Kubernetes Cluster are not suported, as the xref:secret-operator:index.adoc[secret-operator] requires special privileges that are not granted in automatic Kubernetes clusters.

> https://azure.microsoft.com/en-us/products/kubernetes-service

Author: That's just how we did it with other clouds as well in https://docs.stackable.tech/home/nightly/kubernetes/, copied it from there

Basically in case a customer is interested in reading the vendor docs

> https://azure.microsoft.com/en-us/products/kubernetes-service

Author: Should we put the link behind the text "Vendor link"?

> https://azure.microsoft.com/en-us/products/kubernetes-service

Reviewer: Hmm ok if it's like that everywhere it's fine. We can just go in and change it another time with a link maybe

. Create Kubernetes cluster
+
Choose an arbitrary name, the default configurations should be fine.
+
image::managed-k8s/aks/1.png[]

. Update NodePool settings
+

> We recommend checking `Enable public IP per node`, so that our demos work out of the box as they are using NodePorts for maximum portability.

Author: I would prefer to keep the "we recommend", as it's just much much easier. You can still don't expose your cluster publicly and access many (but not all) things via port-forward.

> We recommend checking `Enable public IP per node`, so that our demos work out of the box as they are using NodePorts for maximum portability.

Reviewer: I mainly wanted to get rid of the 'we' and 'our' - maybe the formulation below can work? 

```suggestion
It is recommended to check `Enable public IP per node`, so that the Stackable demos work out of the box as they are using NodePorts for maximum portability.
```

+
image::managed-k8s/aks/2.png[]
+
In case you don't want to spend more money by having dedicated control-plane nodes, remove the `CriticalAddonsOnly` taint with the trash-button to the right, so that this nodes can also run workloads and not only the dataplane.
+
image::managed-k8s/aks/3.png[]

. Enable public access
+
You also need set `Network policy` to `None`, so that your deployed demos are reachable from the Internet.
+
image::managed-k8s/aks/4.png[]

. Create Kubernetes cluster
+
Click on `Review + Create` to finish the creation

. Allow incoming traffic
+

> Even though we made the Kubernetes nodes puclic and have no traffic policy, incoming connections will still be blocked by default.

Reviewer: ```suggestion
Even though the Kubernetes nodes were configured to be public and have no traffic policy, incoming connections will still be blocked by default.
```
